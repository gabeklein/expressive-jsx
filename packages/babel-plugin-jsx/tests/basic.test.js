transform("component", `
  () => {
    <div>Hello World</div>
  }
`);

transform("multiple children", `
  () => {
    <this>
      <div>Foo</div>
      <div>Bar</div>
    </this>
  }
`);

transform("statement pass-thru", `
  () => {
    const [x, setX] = useState();

    useEffect(() => {
      void x;
      setX();
    }, []);
    
    <div />
  }
`)

transform("basic style", `
  () => {
    color: green;

    <this>Hello World</this>
  };
`);