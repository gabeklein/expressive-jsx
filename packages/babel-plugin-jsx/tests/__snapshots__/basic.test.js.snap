// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic style 1`] = `
  () => <div className="element_7nq">Hello World</div>;
  css(\`
  	.element_7nq {
  		color: green;
  	}
  \`);
`;

exports[`component 1`] = `  () => <div>Hello World</div>;`;

exports[`multiple children 1`] = `
  () => (
    <Fragment>
      <div>Foo</div>
      <div>Bar</div>
    </Fragment>
  );
`;

exports[`statement pass-thru 1`] = `
  () => {
    const [x, setX] = useState();
    useEffect(() => {
      void x;
      setX();
    }, []);
  
    return <div />;
  };
`;
